src/hooks/useCheckAuth.js

```diff
@@ -0,0 +1,15 @@
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import authService from "../auth";
+
+export const useCheckAuth = () => {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const headers = authService.isAuthenticated();
+
+    if (!headers) {
+      navigate("/login");
+    }
```

Comment:
Just some syntax sugar

```suggestion
    !authService.isAuthenticated() && navigate("/login")
```